the-collab-lab/tcl-79-smart-shopping-list
# Add CreateShoppingList file to components folder

## Description
This PR adds a CreateShoppingList component that allows the user to create new shopping lists. 


## Related Issue

Closes #4 

## Acceptance Criteria

UI-related tasks:

:ballot_box_with_check: The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.

:ballot_box_with_check: The input that accepts the name of the item has a semantic label element associated with it

:ballot_box_with_check: The user can submit this form with both the mouse and the Enter key

:ballot_box_with_check: When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

:ballot_box_with_check: Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.

:ballot_box_with_check: The shopping list path is stored in local storage using the setListPath function.

:ballot_box_with_check: Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes

feature

## Updates

### Before

![image](https://github.com/user-attachments/assets/a30c1bb7-2f23-4020-9a61-386bfe4ee313)


### After

![image](https://github.com/user-attachments/assets/48f2279a-d4ab-43e0-a8e1-527a8dba1f2b)
![image](https://github.com/user-attachments/assets/d5fe417b-6387-47b1-bc15-d2c440fbd888)


## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout jo-hm-createList`

- Open the homepage by running `npm start`.

- Enter a shopping list name into the input box.

- Either click 'Create Shopping List' or press the enter key.

- App will then automatically navigate to 'List' if the shopping list has been added successfully and a notification will appear telling the user this.

- If their has been an error adding the shopping list an error notification will appear asking the user to try again.
